tools/run_tests/dockerize/build_interop_image.sh

```diff
@@ -37,6 +37,7 @@ set -x
 #  INTEROP_IMAGE - name of tag of the final interop image
 #  BASE_NAME - base name used to locate the base Dockerfile and build script
 #  TTY_FLAG - optional -t flag to make docker allocate tty
+#  RUNTIME - runtime setting. If specified, Dockerfile-$RUNTIME will be used.
```

Comment:
Yes, I am suggesting to flatten the configurations into what you call "languages" - but it's just different dockerfiles.   What is a "language" and "runtime" is basically something that go/grpc-matrix can determine according to your needs.  It just need to know that what is under "Java" and "OpenJDK 8"  corresponds to docker image "grpc_interop_java_openjdk8".  Not a big deal, and you don't need to propagate these concepts all the way to the level of run_interop_tests.py (or at least not to the level of build_interop_image shell scripts).